qa/L0_python_api/testing_utils.py

```diff
@@ -73,16 +75,31 @@ def teardown_service(service: Union[KServeHttp, KServeGrpc]) -> None:
     service.stop()
 
 
-def setup_client(frontend_client, url: str):
+def setup_client(
+    frontend_client: Union["tritonclient.http", "tritonclient.grpc"], url: str
+):
+    """
+    Sets up a client to communicate with the Server through the respective protocol.
+    """
     return frontend_client.InferenceServerClient(url=url)
 
 
-def teardown_client(client) -> None:
+def teardown_client(
+    client: Union[
+        "tritonclient.http.InferenceServerClient",
+        "tritonclient.grpc.InferenceServerClient",
```

Comment:
I think this type hint is correct, and the other places you use `Union` is missing `InferenceServerClient`. You could probably also use `InferenceServerClientBase` though it'd be a bit less strict.